packages/core/src/codewhisperer/commands/startSecurityScan.ts

```diff
@@ -250,7 +250,11 @@ export async function startSecurityScan(
                 CodeScansState.instance.setMonthlyQuotaExceeded()
             }
         }
-        codeScanTelemetryEntry.reason = (error as SecurityScanError).message
+        if ((error as ToolkitError).code == 'ContentLengthError') {
+            codeScanTelemetryEntry.reason = 'Payload size limit reached'
+        } else {
+            codeScanTelemetryEntry.reason = (error as SecurityScanError).message
```

Comment:
`reason` is for code-like strings. Use `reasonDesc` for free-form text. use `getTelemetryReasonDesc` to get it: https://github.com/aws/aws-toolkit-vscode/blob/cb603edfe2d2429ff4ee080ae4094a7de2806989/packages/core/src/shared/errors.ts#L310

```suggestion
            codeScanTelemetryEntry.reasonDesc = getTelemetryReasonDesc(error)
```

Reply:
Updated accordingly.